chapters/problems/sections/terrain_problems.tex

```diff
@@ -0,0 +1,10 @@
+\section{Terrain problems} \label{sec:terrain_problems}
+The main problem with terrain was performance.
+Generating terrain is not a particularly resource consuming task, however editing the terrain is.
+The main challenge was making editing the terrain smooth.
+To achieve that the terrain has to be updated at least 30 times per second which means that multiple chunks, each one having $19^3$ vertices have to be updated at least 30 times per second.
+This process was so resource consuming that it had to be moved to a separate thread as described in \autoref{sec:chunk-worker}.
+However since the main thread is the one that communicates with the GPU and the physics engine, the main thread still had to be involved in the process.
+This created many problems connected with making sure that there are no race conditions while maintaining performance.
+To resolve these problems we used a system of locks, concurrent queues and concurrent hashsets.
```

Comment:
maybe don't mention locks, etc here, just give a ref
This is specifically because I'm not even sure if we still need locks to be honest (I didn't remove them after we added batch processing because they don't cause any harm but who knows if they are necessary)